nexus_id_spec.md

```diff
@@ -0,0 +1,147 @@
+# Nexus Channel Core - Nexus ID Specification
+
+Version: 0.9.0
+
+## Purpose and Background
+
+The Nexus ID is a globally-unique identifier that is *required* to be stored
+by all devices that implement Nexus Channel Core. It solves many challenges
+facing manufacturers of interoperable devices that may need to be managed by
+external software platforms.
+
+## Structure
+
+The Nexus ID is a 6-byte structure that consists of two parts:
+
+```
+ <-2 bytes-> <----4 bytes---->
++-----------+-----------------+
+| Authority |    Device ID    |
+| ID        |                 |
++-----------+-----------------+
+```
+
+* 2-byte 'Authority ID' indicating what party assigned the device ID
+* 4-byte 'Device ID' unique for all IDs assigned from a given authority
+
+Manufacturers are assigned a unique Authority ID by the maintainer of Nexus
+(currently Angaza). Manufacturers can then serialize a practically infinite
+number of Device IDs according to their own practices and preferences. 
+
+As long as all manufacturers guarantee that they are using their own Authority 
+ID and that there are no collisions within their own Device ID space, then there 
+cannot be any identity collisions between devices communicating on the same Nexus
+Channel Core network.
+
+## External vs. Internal Identifiers
+
+Many device management software platforms require a device-specific identifier
+to reference value-added services related to that device. For example, Angaza 
+uses a device's identifier to reference its related PAYG loan account.
+
+The Device ID part of the Nexus ID *can* be used as an external identifier for
+these purposes, if a manufacturer wishes to reduce the number of serialization
+schemes in use. However, it is not required that the Nexus ID is exposed
+externally. It must just be assigned and stored internally in the device
+firmware (and accessed through Nexus Channel Core APIs) to avoid addressing 
```

Comment:
Note: implementers of this spec *might not* use our reference implementation code, so the part about the Nexus Channel Core APIs might not be necessary.

Reply:
Ah, by this I was specifically referring to `nxp_channel_get_nexus_id` -- just a way to get the stored Nexus ID. Every Nexus Channel Core device *must* have this implemented right?

Reply:
Not every Nexus Channel Core device will have any code from `nexus-embedded` (e.g. I imagine someone can build a totally independent system without our reference C code at any level that still interoperates with our system, if they follow the spec). So, in that case, they'd have no `nx` or `nxp` functions at all, but would just need a way to store and access a Nexus ID.

Reply:
Oh I see, I totally get it now. OK, will remove that bit.